internal/validators/oci_validator.go

```diff
@@ -0,0 +1,300 @@
+	details := make([]string, 0)
+
+	reg, err := name.NewRegistry(host)
+	if err != nil {
+		details = append(details, fmt.Sprintf("failed to get registry %v", host))
+		return details, err
+	}
+
+	repoList, err := remote.Catalog(context.Background(), reg, opts...)
+	if err != nil {
+		details = append(details, fmt.Sprintf("failed to list repositories in registry %v", host))
+		return details, err
+	}
+
+	if len(repoList) == 0 {
+		details = append(details, fmt.Sprintf("no repositories found in registry %v", host))
+		return details, nil
+	}
+
+	var repo name.Repository
+	var ref name.Reference
+	for _, curRepo := range repoList {
+		repo, err = name.NewRepository(host + "/" + curRepo)
+		if err != nil {
+			details = append(details, fmt.Sprintf("failed to get repository %v/%v", host, curRepo))
+			continue
+		}
+
+		tags, err := remote.List(repo, opts...)
+		if err != nil {
+			details = append(details, fmt.Sprintf("failed to get tags for repository %v/%v", host, curRepo))
+			continue
+		}
+
+		if len(tags) == 0 {
+			details = append(details, fmt.Sprintf("no tags found in repository %v/%v", host, curRepo))
+			continue
+		}
+
+		tag := tags[0]
+		ref, err = generateRef(host, fmt.Sprintf("%v:%v", curRepo, tag), vr)
+		if err != nil {
+			details = append(details, fmt.Sprintf("failed to generate reference for artifact %v/%v:%v", host, curRepo, tag))
+			continue
+		}
+		break
+	}
+	if err != nil {
+		return details, err
+	}
+
+	// download image without storing it on disk
+	_, err = remote.Image(ref, opts...)
+	if err != nil {
+		details = append(details, fmt.Sprintf("failed to download artifact %v", ref.Name()))
+		return details, err
+	}
+
+	return []string{}, nil
+}
+
+func generateRef(registry, artifact string, vr *types.ValidationResult) (name.Reference, error) {
+	if strings.Contains(artifact, "@sha256:") {
+		return name.NewDigest(fmt.Sprintf("%s/%s", registry, artifact))
+	}
+
+	if !strings.Contains(artifact, ":") {
+		vr.Condition.Details = append(vr.Condition.Details, fmt.Sprintf("artifact %v does not contain a tag or digest, defaulting to \"latest\" tag", artifact))
+	}
+
+	return name.NewTag(fmt.Sprintf("%s/%s", registry, artifact))
+}
+
+func getECRLoginToken(username, password, region string) (string, error) {
+	cfg, err := config.LoadDefaultConfig(
+		context.Background(),
+		config.WithRegion(region),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(username, password, "")),
+	)
+	if err != nil {
+		return "", fmt.Errorf("failed to load configuration, %v", err)
+	}
+
+	client := ecr.NewFromConfig(cfg)
+
+	output, err := client.GetAuthorizationToken(context.TODO(), &ecr.GetAuthorizationTokenInput{})
+	if err != nil {
+		return "", fmt.Errorf("failed to get ECR authorization token: %v", err)
+	}
+
+	for _, data := range output.AuthorizationData {
+		return *data.AuthorizationToken, nil
+	}
+
+	return "", fmt.Errorf("no authorization data available")
+}
+
+func parseEcrRegion(URL string) (string, error) {
```

Comment:
nit: inconsistent casing. please rename to `parseECRRegion` or rename above func to `getEcrLoginToken`. My personal preference would be the latter.

Reply:
Additionally, the `URL` param should be `url`

Reply:
And "Invalid ecr url %v" should be "Invalid ECR URL %v"